pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -125,6 +125,9 @@ func Execute(configFile io.Reader) {
 
 	// Registry extensions endpoint provides prometheus metrics.
 	if extraConfig.Metrics.Enabled {
+		if len(extraConfig.Metrics.Secret) == 0 {
+			context.GetLogger(app).Fatalf("secret token for metrics endpoint cannot be empty")
```

Comment:
maybe specify the path like: `the openshift.metrics.secret field cannot be empty when metrics are enabled`